src/components/vm/overview/vsock.jsx

```diff
@@ -0,0 +1,259 @@
+            vmName: vm.name,
+            hotplug: vm.state === "running",
+            permanent: vm.persistent,
+        })
+                .then(Dialogs.close, exc => setDialogError({ text: _("Failed to detach vsock"), detail: exc.message }))
+                .finally(() => setInProgress(false));
+    };
+
+    const showWarning = () => {
+        if (isVsockAttached && vm.persistent && vm.state === "running" &&
+            (vmVsockNormalized.auto !== auto ||
+            // If automatic generation is set, then adress in live XML is prefilled with a value libvirt chooses,
+            // and it's expected that live XML will contain different address than inactiveXML
+            (!auto && vmVsockNormalized.address !== address)))
+            return <WarningInactiveAlert idPrefix={idPrefix} />;
+    };
+
+    const vsockUsage = getVsockUsageMessage(vm.name, vm.connectionName, vms, auto, address);
+
+    const body = (
+        <Form onSubmit={e => e.preventDefault()} isHorizontal>
+            <FormGroup fieldId="vsock-cid"
+                       label={_("Context identifier")}
+                       hasNoPaddingTop
+                       isInline>
+                <Radio id="vsock-cid-generate"
+                       name="generate"
+                       isChecked={auto}
+                       label={ASSIGN_AUTOMATICALLY}
+                       onChange={() => setAuto(true)} />
+                <Radio id="vsock-cid-set"
+                       name="set"
+                       isChecked={!auto}
+                       label={SET_MANUALLY}
+                       onChange={() => setAuto(false)} />
+            </FormGroup>
+            <FormGroup fieldId="vsock-cid"
+                       helperText={vsockUsage}
+                       validated="warning"
+                       hasNoPaddingTop
+                       hasN
+                       isInline>
+                <NumberInput value={!auto ? address : ""}
+                             onMinus={() => setAddress(address - 1)}
+                             onChange={event => setAddress(event.target.value)}
+                             onPlus={() => setAddress(address + 1)}
+                             onBlur={event => onBlur(event.target.value)}
+                             min={MIN_VSOCK_CID}
+                             isDisabled={auto}
+                             inputName="vsock-context-identifier"
+                             id="vsock-context-identifier"
+                             inputAriaLabel="vsock context identifier"
+                             allowEmptyInput
+                             widthChars={10} />
+            </FormGroup>
+        </Form>
+    );
+
+    return (
+        <Modal id={`${idPrefix}-vsock-modal`}
+               position="top"
+               variant="medium"
+               onClose={Dialogs.close}
+               title={isVsockAttached ? _("Edit vsock interface") : _("Add vsock interface")}
```

Comment:
vsock vs. Vsock -- let's please pick one spelling.

Reply:
Unresolving: 
This doesn't look right. Here every entry should start with uppercase. So for overview tab, it should be `Vsock`
![Uploading TestMachinesLifecycle-testBasic-vm-details-pixels.png…]()

Reply:
The description list now uses ["Vsock"](https://github.com/cockpit-project/cockpit-machines/pull/969/commits/d814abbd82597355dd008a74ae2c07b4194e25b3#diff-70cd150703f2ad775366f80db32dbd98dbeffb82408df0f1cb255aacafe87777R229), so that should be resolved now?